## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#58dfd673-cda8-4f2d-a201-5929821fe101) |
| mint-large-bucket.sh | :x: [more...](#5376217a-57b5-4e85-a9e1-2d1648516fbb) |
| mint-dist-xl.sh | :x: [more...](#2e13f2b1-bd03-4741-a473-e320ddf65181) |
<a name="58dfd673-cda8-4f2d-a201-5929821fe101"></a>
### 9921-cb86e46/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30044
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c8d8aa59136d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 25 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 58 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 57 seconds
{
  "args": {
    "bucketName": "minio-go-test-kk01lifthftdcdar",
    "objectName": "fehzsxx30w3gp3fog0hk3wjq1p4sbg",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 10151,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 2 minutes and 54 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 359,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-3gcrv7m, objectName = null, resource = /minio-java-test-3gcrv7m, requestId = 161C17D86F037F60, hostId = d0ea1e4b-860e-4a51-9f83-ad5442bc25f5)\nrequest={method=DELETE, url=http://minio-dev6.minio.io:30044/minio-java-test-3gcrv7m, headers=Host: minio-dev6.minio.io:30044\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200626T122946Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200626/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-06-26T12-14-29Z\nVary: Origin\nX-Amz-Request-Id: 161C17D86F037F60\nX-Xss-Protection: 1; mode=block\nDate: Fri, 26 Jun 2020 12:29:46 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executeDelete(MinioClient.java:1366), io.minio.MinioClient.removeBucket(MinioClient.java:3686), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3314), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(9/15) Running minio-js tests ... done in 47 seconds
(10/15) Running minio-py tests ... done in 1 minutes and 44 seconds
(11/15) Running s3cmd tests ... done in 22 seconds
(12/15) Running s3select tests ... done in 3 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="5376217a-57b5-4e85-a9e1-2d1648516fbb"></a>
### 9921-cb86e46/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31783
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c9ca3d7a6e71:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-vmrkn39znh55d6",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 121055,
  "error": "BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.\n\tstatus code: 409, request id: 161C178D16948E26, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 5 seconds
(2/15) Running aws-sdk-php tests ... FAILED in 8 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-vmrkn39znh55d6",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 121055,
  "error": "BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.\n\tstatus code: 409, request id: 161C178D16948E26, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(2/15) Running aws-sdk-ruby tests ... FAILED in 26 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-1bd5a4787ad1"
  },
  "duration": 11516.52,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-1bd5a4787ad1', The bucket you tried to delete is not empty",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 9780,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-12603/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-12603 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(2/15) Running healthcheck tests ... done in 1 seconds
(3/15) Running mc tests ... FAILED in 6 seconds
{
  "name": "mc",
  "duration": "655",
  "function": "setup",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-28218 --quiet --no-color mb myminio/mc-test-bucket-17257  >>> \nmc: <ERROR> Unable to make bucket `myminio/mc-test-bucket-17257`. Your previous request to create the named bucket succeeded and you already own it."
}
(3/15) Running minio-dotnet tests ... done in 4 minutes and 29 seconds
(4/15) Running minio-go tests ... FAILED in 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-du6v45uk16lk1hg1",
    "objectName": "etz1aydip435hp4n6i6pavb2n22rsz",
    "region": "eu-west-1"
  },
  "duration": 4145,
  "error": "The bucket you tried to delete is not empty",
  "function": "MakeBucket(bucketName, region)",
  "message": "Cleanup failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(4/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 26,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-2kt1btb, objectName = null, resource = /minio-java-test-2kt1btb, requestId = 161C17DD44948BDE, hostId = 1f6698df-c964-45cd-8cfb-e262eb689113)\nrequest={method=PUT, url=http://minio-c3.minio.io:31783/minio-java-test-2kt1btb, headers=Host: minio-c3.minio.io:31783\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200626T123007Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200626/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 305\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-26T12-14-29Z\nVary: Origin\nX-Amz-Request-Id: 161C17DD44948BDE\nX-Xss-Protection: 1; mode=block\nDate: Fri, 26 Jun 2020 12:30:07 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(4/15) Running minio-js tests ... FAILED in 15 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(4/15) Running minio-py tests ... FAILED in 2 minutes and 4 seconds
{
  "name": "minio-py:test_list_buckets",
  "function": "list_buckets()",
  "duration": 120133,
  "message": "BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 270, in test_list_buckets\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 374, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.BucketAlreadyOwnedByYou: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1979, in main\n    test_list_buckets(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 279, in test_list_buckets\n    raise Exception(err)\nException: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n",
  "status": "FAIL"
}
(4/15) Running s3cmd tests ... done in 40 seconds
(5/15) Running s3select tests ... FAILED in 24 minutes and 16 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-1f1449d8-b39c-4dd7-8c9b-4d226582b2ca"
  },
  "duration": 1455131,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(5/15) Running security tests ... done in 0 seconds

Executed 5 out of 15 tests successfully.
```
<a name="2e13f2b1-bd03-4741-a473-e320ddf65181"></a>
### 9921-cb86e46/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30974
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e61733292027:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 minutes and 9 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0muts3f56asnpp",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 125230,
  "error": "BucketNotEmpty: The bucket you tried to delete is not empty\n\tstatus code: 409, request id: 161C178DBBAB40AF, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 5 seconds
(2/15) Running aws-sdk-php tests ... done in 1 minutes and 11 seconds
(3/15) Running aws-sdk-ruby tests ... FAILED in 13 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-a587ddfd5397"
  },
  "duration": 5430.13,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-a587ddfd5397', The bucket you tried to delete is not empty",
  "status": "FAIL"
}
(3/15) Running awscli tests ... FAILED in 22 seconds
{
  "name": "awscli",
  "duration": 16128,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-18716/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-18716 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(3/15) Running healthcheck tests ... done in 0 seconds
(4/15) Running mc tests ... done in 1 minutes and 9 seconds
(5/15) Running minio-dotnet tests ... done in 2 minutes and 8 seconds
(6/15) Running minio-go tests ... FAILED in 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-wrtdm0hjb4itb06m",
    "objectName": "qxgauhjn4witcllg191uwq6qwe4ste",
    "region": "eu-west-1"
  },
  "duration": 2712,
  "error": "The bucket you tried to delete is not empty",
  "function": "MakeBucket(bucketName, region)",
  "message": "Cleanup failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(6/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 26,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-18tidrs, objectName = null, resource = /minio-java-test-18tidrs, requestId = 161C17D716D3CB93, hostId = 6f83a5dd-ac19-4b39-b750-ef7227bfb26d)\nrequest={method=PUT, url=http://minio-dev4.minio.io:30974/minio-java-test-18tidrs, headers=Host: minio-dev4.minio.io:30974\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200626T122940Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200626/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 305\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-26T12-14-29Z\nVary: Origin\nX-Amz-Request-Id: 161C17D716D3CB93\nX-Xss-Protection: 1; mode=block\nDate: Fri, 26 Jun 2020 12:29:40 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(6/15) Running minio-js tests ... FAILED in 44 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 7,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(6/15) Running minio-py tests ... FAILED in 6 minutes and 2 seconds
{
  "name": "minio-py:test_fput_object_small_file",
  "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata, sse, progress, part_size)",
  "args": {
    "bucket_name": "minio-py-test-4591e131-c1bf-4a39-98b5-153a8ed52c7a",
    "object_name": "abd79f05-edb2-4210-bace-e56d3be01d7b",
    "file_path": "/mint/data/datafile-1-MB",
    "metadata": {
      "x-amz-storage-class": "STANDARD_IA"
    }
  },
  "duration": 240724,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 371, in test_fput_object_small_file\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1983, in main\n    test_fput_object_small_file(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 373, in test_fput_object_small_file\n    raise Exception(err)\nException: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(6/15) Running s3cmd tests ... FAILED in 21 seconds
{
  "name": "s3cmd",
  "duration": "599",
  "function": "teardown",
  "status": "FAIL",
  "error": "WARNING: Bucket is not empty. Removing all the objects from it first. This may take some time...\nERROR: Parameter problem: Expecting S3 URI with a filename or --recursive: s3://s3cmd-test-bucket-9547/"
}
(6/15) Running s3select tests ... FAILED in 38 minutes and 4 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-2451fa15-4578-4179-b1c4-8179dac9d83d"
  },
  "duration": 2283053,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/pkg/s3select/internal/parquet-go/decode.go
+++ b/pkg/s3select/internal/parquet-go/decode.go
@@ -267,8 +267,11 @@ func readRLE(reader *bytes.Reader, header, bitWidth uint64) (result []int64, err
 	}
 
 	val := int64(bytesToUint32(data))
-
 	count := header >> 1
+	if count > math.MaxInt64/8 {
+		// 8 bytes/element.
+		return nil, errors.New("parquet: size too large")
+	}
 	result = make([]int64, count)
 	for i := range result {
 		result[i] = val
@@ -441,7 +444,7 @@ func readDataPageValues(
 		if err != nil {
 			return nil, -1, err
 		}
-		if len(i64s) < int(count) || count > math.MaxInt64 {
+		if len(i64s) < int(count) || count > math.MaxInt64/8 {
 			return nil, -1, errors.New("parquet: value out of range")
 		}
 		return i64s[:count], parquet.Type_INT64, nil
@@ -452,7 +455,7 @@ func readDataPageValues(
 			return nil, -1, err
 		}
 
-		if len(i64s) < int(count) || count > math.MaxInt64 {
+		if len(i64s) < int(count) || count > math.MaxInt64/8 {
 			return nil, -1, errors.New("parquet: value out of range")
 		}
 		i64s = i64s[:count]
@@ -472,7 +475,7 @@ func readDataPageValues(
 			return nil, -1, err
 		}
 
-		if len(i64s) < int(count) || count > math.MaxInt64 {
+		if len(i64s) < int(count) || count > math.MaxInt64/8 {
 			return nil, -1, errors.New("parquet: value out of range")
 		}
 		i64s = i64s[:count]
@@ -488,7 +491,7 @@ func readDataPageValues(
 		if err != nil {
 			return nil, -1, err
 		}
-		if len(byteSlices) < int(count) || count > math.MaxInt64 {
+		if len(byteSlices) < int(count) || count > math.MaxInt64/24 {
 			return nil, -1, errors.New("parquet: value out of range")
 		}
 
@@ -499,7 +502,7 @@ func readDataPageValues(
 		if err != nil {
 			return nil, -1, err
 		}
-		if len(byteSlices) < int(count) || count > math.MaxInt64 {
+		if len(byteSlices) < int(count) || count > math.MaxInt64/24 {
 			return nil, -1, errors.New("parquet: value out of range")
 		}
 

--- a/pkg/s3select/internal/parquet-go/page.go
+++ b/pkg/s3select/internal/parquet-go/page.go
@@ -217,11 +217,11 @@ func readPage(
 				return nil, 0, 0, err
 			}
 
-			if repetitionLevels = values.([]int64); uint64(len(repetitionLevels)) > numValues {
+			if repetitionLevels = values.([]int64); len(repetitionLevels) > int(numValues) && int(numValues) >= 0 {
 				repetitionLevels = repetitionLevels[:numValues]
 			}
 		} else {
-			if numValues > math.MaxInt64 {
+			if numValues > math.MaxInt64/8 {
 				return nil, 0, 0, errors.New("parquet: numvalues too large")
 			}
 			repetitionLevels = make([]int64, numValues)
@@ -234,10 +234,16 @@ func readPage(
 			if err != nil {
 				return nil, 0, 0, err
 			}
-			if definitionLevels = values.([]int64); uint64(len(definitionLevels)) > numValues {
+			if numValues > math.MaxInt64/8 {
+				return nil, 0, 0, errors.New("parquet: numvalues too large")
+			}
+			if definitionLevels = values.([]int64); len(definitionLevels) > int(numValues) {
 				definitionLevels = definitionLevels[:numValues]
 			}
 		} else {
+			if numValues > math.MaxInt64/8 {
+				return nil, 0, 0, errors.New("parquet: numvalues too large")
+			}
 			definitionLevels = make([]int64, numValues)
 		}
 

